lib/cylc/cycling/iso8601.py

```diff
@@ -791,33 +825,97 @@ def _get_old_strptime_format(point_string):
         return None
 
 
-def test():
-    """Run some simple tests for date-time cycling."""
-    cylc_config = {"cylc": {"cycle point num expanded year digits": 0,
-                            "cycle point format": None,
-                            "cycle point time zone": None}}
-    init_from_cfg(cylc_config)
-    p_start = ISO8601Point('20100808T00')
-    p_stop = ISO8601Point('20100808T02')
-    i = ISO8601Interval('PT6H')
-    print p_start - i
-    print p_stop + i
-
-    print
-    r = ISO8601Sequence('PT10M', str(p_start), str(p_stop),)
-    r.set_offset(- ISO8601Interval('PT10M'))
-    p = r.get_next_point(ISO8601Point('20100808T0000'))
-    print p
-    while p and p < p_stop:
-        print ' + ' + str(p), r.is_on_sequence(p)
-        p = r.get_next_point(p)
-    print
-    while p and p >= p_start:
-        print ' + ' + str(p), r.is_on_sequence(p)
-        p = r.get_prev_point(p)
-
-    print
-    print r.is_on_sequence(ISO8601Point('20100809T0005'))
+class TestISO8601Sequence(unittest.TestCase):
+    """Contains unit tests for the ISO8601Sequence class."""
+
+
+    def test_exclusions_simple(self):
+        """Test the generation of points for sequences with exclusions."""
+        init()
+        sequence = ISO8601Sequence('PT1H!20000101T02Z', '20000101T00Z')
+
+        output = []
+        point = sequence.get_start_point()
+        count = 0
+        while point and count < 4:
+            output.append(point)
+            point = sequence.get_next_point(point)
+            count += 1
+        output = [str(out) for out in output]
+        self.assertEqual(output, ['20000101T0100+01', '20000101T0200+01',
```

Comment:
To get these in `Z` (and hence passing elsewhere) you need to set up the time zone in the configuration singleton in this file.